src/framework/app-base.js

```diff
@@ -2114,6 +2114,12 @@ const makeTick = function (_app) {
         if (!application.graphicsDevice)
             return;
 
+        // cancel any hanging rAF to avoid multiple rAF callbacks per frame
+        if (application.frameRequestId) {
+            application.xr?.session?.cancelAnimationFrame(application.frameRequestId);
+            cancelAnimationFrame(application.frameRequestId);
+        }
+
```

Comment:
Should the `application.frameRequestId = null;` line below move inside the `if`?